tensorflow/python/tools/optimize_for_inference_lib.py

```diff
@@ -405,19 +424,24 @@ def fuse_resize_and_conv(input_graph_def):
       new_ops.extend([paddings_op])
       mirror_paddings_name = paddings_op.name
       mirror_paddings_mode = tf.AttrValue(s=b"REFLECT")
-    fused_conv_op.input.extend([resize_op.input[0], resize_op.input[1],
-                                mirror_paddings_name, conv_op.input[1]])
+    if resize_op:
+      fused_conv_op.input.extend([resize_op.input[0], resize_op.input[1],
+                                  mirror_paddings_name, conv_op.input[1]])
+      fused_conv_op.attr["resize_align_corners"].CopyFrom(
+          resize_op.attr["align_corners"])
+    else:
+      fused_conv_op.input.extend([mirror_pad_op.input[0], mirror_paddings_name,
+                                  conv_op.input[1]])
     fused_conv_op.attr["T"].CopyFrom(conv_op.attr["T"])
-    fused_conv_op.attr["resize_align_corners"].CopyFrom(
-        resize_op.attr["align_corners"])
     fused_conv_op.attr["mode"].CopyFrom(mirror_paddings_mode)
     fused_conv_op.attr["strides"].CopyFrom(conv_op.attr["strides"])
     fused_conv_op.attr["padding"].CopyFrom(conv_op.attr["padding"])
     new_ops.extend([fused_conv_op])
 
   result_graph_def = tf.GraphDef()
   for node in input_graph_def.node:
-    if node.name in nodes_to_skip:
+    if (node.name in node_reference_count.keys() and
```

Comment:
It did change the logic a bit, because output nodes were never mentioned in the inputs of other nodes (so had no key) but we wanted to keep them. I like the cleanliness of this approach though, so I switched to explicitly passing in the names of the outputs.